### ISSUE TITLE
hiredis/adapters/libevent.h memleak

### ISSUE THREAD

**Issue by postren** [redis/hiredis#618]
Hi,
in function `redisLibeventAttach` 
`    /* Initialize and install read/write events */
    e->rev = event_new(base, c->fd, EV_READ, redisLibeventReadEvent, e);
    e->wev = event_new(base, c->fd, EV_WRITE, redisLibeventWriteEvent, e);
`

here `event_new` will **memleak**

please update as flollow:
```static void redisLibeventCleanup(void *privdata) {
    redisLibeventEvents *e = (redisLibeventEvents*)privdata;
    event_del(e->rev);
    event_del(e->wev);
    event_free(e->rev);
    event_free(e->wev);
    free(e);
}

```

**Comment by michael-grunder** [comment]
Hi @postren I'm going through old issues.

I think this was fixed when timeouts were added in 35a0a1f3.

I tested with a [small example program](https://gist.github.com/michael-grunder/7977edaafa54d8c07e4a9f1509bdb3d6) and don't see any leaks:

```bash
==31950== Memcheck, a memory error detector
==31950== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==31950== Using Valgrind-3.16.0.GIT and LibVEX; rerun with -h for copyright info
==31950== Command: ./a.out
==31950==
Connected (no timeout change)
 OK: 763
Disconnected
==31950==
==31950== HEAP SUMMARY:
==31950==     in use at exit: 0 bytes in 0 blocks
==31950==   total heap usage: 92 allocs, 92 frees, 26,196 bytes allocated
==31950==
==31950== All heap blocks were freed -- no leaks are possible
==31950==
==31950== For lists of detected and suppressed errors, rerun with: -s
==31950== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```

**Comment by postren** [comment]
@michael-grunder Thanks.
